tools/statplot.py

```diff
@@ -237,50 +272,74 @@ def PlotData(self, action, type, xscale, yscale):
-                self.yCombo.set_active(1)
-        elif action == 'reload':
-            self.xCombo.set_active_iter(prevIterX)
-            self.yCombo.set_active_iter(prevIterY)
+    def ReadData(self):
+        def GatherEntries(stat2read):
+            print('Reading ' + stat2read)
+            entries = numpy.array([])
+            with open(stat2read, 'r') as fid:
+                for num, line in enumerate(fid):
+                    if line.startswith('<field'):
+                        info = etree.fromstring(line)
+                        if info.get('statistic') == 'value':
+                            entries = numpy.append(entries, info.get('name'))
+                        elif all([x in info.keys() for x in
+                                  ['components', 'material_phase']]):
+                            for i in range(int(info.get('components'))):
+                                entries = numpy.append(
+                                    entries, info.get('material_phase') + '%'
+                                    + info.get('name') + '%'
+                                    + info.get('statistic') + f'%{i}')
+                        elif 'components' in info.keys():
+                            for i in range(int(info.get('components'))):
+                                entries = numpy.append(
+                                    entries, info.get('name') + '%'
+                                    + info.get('statistic') + f'%{i}')
+                        elif 'material_phase' in info.keys():
+                            entries = numpy.append(
+                                entries, info.get('material_phase') + '%'
+                                + info.get('name') + '%'
+                                + info.get('statistic'))
+                        else:
+                            entries = numpy.append(
+                                entries, info.get('name') + '%'
+                                + info.get('statistic'))
+                    elif line.startswith('</header>'):
+                        break
+            return entries, num
 
-    def ReadData(self, statfile):
-        stats = []
-        for i, filename in enumerate(statfile):
-            failcount = 0
-            while failcount < 4:
-                try:
-                    stats.append(fluidity_tools.Stat(filename))
-                    break
-                except (TypeError, ValueError):
-                    time.sleep(0.2)
-                    failcount += 1
-            if failcount == 4:
-                stats.append(fluidity_tools.Stat(filename))
-        if len(stats) == 1:
-            return stats[0]
-        else:
-            return fluidity_tools.JoinStat(*stats)
+        if self.statfile[0].endswith('stat'):
+            entries, num = GatherEntries(self.statfile[0])
+            values = numpy.genfromtxt(self.statfile[0], skip_header=num + 1)
+            values = values.T[numpy.argsort(entries)].T
```

Comment:
Can you just index along the second dimension instead of double-transposing (`values[:, numpy.argsort(entries)]`)?

Reply:
True, thank you.